lib/runtime/invalid_transaction.go

```diff
@@ -0,0 +1,146 @@
+	errMandatoryDispatchError   = errors.New("mandatory dispatch error")
+	errInvalidMandatoryDispatch = errors.New("invalid mandatory dispatch")
+)
+
+// Call The call of the transaction is not expected
+type Call struct{}
+
+// Index Returns VDT index
+func (Call) Index() uint { return 0 }
+
+// Payment General error to do with the inability to pay some fees (e.g. account balance too low)
+type Payment struct{}
+
+// Index Returns VDT index
+func (Payment) Index() uint { return 1 }
+
+// Future General error to do with the transaction not yet being valid (e.g. nonce too high)
+type Future struct{}
+
+// Index Returns VDT index
+func (Future) Index() uint { return 2 }
+
+// Stale General error to do with the transaction being outdated (e.g. nonce too low)
+type Stale struct{}
+
+// Index Returns VDT index
+func (Stale) Index() uint { return 3 }
+
+// BadProof General error to do with the transaction’s proofs (e.g. signature)
+type BadProof struct{}
+
+// Index Returns VDT index
+func (BadProof) Index() uint { return 4 }
+
+// AncientBirthBlock The transaction birth block is ancient
+type AncientBirthBlock struct{}
+
+// Index Returns VDT index
+func (AncientBirthBlock) Index() uint { return 5 }
+
+// ExhaustsResources The transaction would exhaust the resources of current block
+type ExhaustsResources struct{}
+
+// Index Returns VDT index
+func (ExhaustsResources) Index() uint { return 6 }
+
+var invalidCustom InvalidCustom
+
+// InvalidCustom Any other custom invalid validity that is not covered
+type InvalidCustom uint8
+
+// Index Returns VDT index
+func (InvalidCustom) Index() uint { return 7 }
+
+// BadMandatory An extrinsic with a Mandatory dispatch resulted in Error
+type BadMandatory struct{}
+
+// Index Returns VDT index
+func (BadMandatory) Index() uint { return 8 }
+
+// MandatoryDispatch A transaction with a mandatory dispatch
+type MandatoryDispatch struct{}
+
+// Index Returns VDT index
+func (MandatoryDispatch) Index() uint { return 9 }
+
+// Set will set a VaryingDataTypeValue using the underlying VaryingDataType
+func (i *InvalidTransaction) Set(val scale.VaryingDataTypeValue) (err error) {
+	vdt := scale.VaryingDataType(*i)
+	err = vdt.Set(val)
+	if err != nil {
+		return err
+	}
+	*i = InvalidTransaction(vdt)
+	return nil
+}
+
+// Value will return value from underying VaryingDataType
+func (i *InvalidTransaction) Value() (val scale.VaryingDataTypeValue) {
+	vdt := scale.VaryingDataType(*i)
+	return vdt.Value()
+}
+
+// NewInvalidTransaction is constructor for InvalidTransaction
+func NewInvalidTransaction() InvalidTransaction {
+	vdt, err := scale.NewVaryingDataType(Call{}, Payment{}, Future{}, Stale{}, BadProof{}, AncientBirthBlock{},
+		ExhaustsResources{}, invalidCustom, BadMandatory{}, MandatoryDispatch{})
+	if err != nil {
+		panic(err)
+	}
+	return InvalidTransaction(vdt)
+}
+
+func (i *InvalidTransaction) Error() string {
+	switch val := i.Value().(type) {
+	case Call:
+		return errUnexpectedTxCall.Error()
```

Comment:
You could have each of these error strings returned from a method `Error() string` on `type Call struct{}`. That way you could just do:

```go
type errorer interface {
  Error() string
}
value, ok := i.Value().(errorer)
if !ok {
  // we are missing the Error() string method on the type
  panic(...)
}

return value.Error()
```

And each error string would be more local to the actually i.Value() type 🤔

Reply:
I think that the `type errorer interface` is not needed as `error` by itself is an interface, so:
```go 
value, ok := i.Value().(error)
```
would be enough to cast the type to an error

Reply:
I just got rid of these errors and returned a string instead. So don't think this is necessary anymore, but let me know if that is not the case.